projects/imagemagick/build.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash -eu
+# Copyright 2017 Google Inc.  #
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# move the corpus
+mkdir afl_testcases
+(cd afl_testcases; tar xvf "$SRC/afl_testcases.tgz")
+mkdir corpus
+find afl_testcases -type f -exec mv -n {} corpus/ \;
+zip -rj afl_testcases.zip corpus/
+
+./configure --prefix="$WORK" --disable-shared --disable-docs
+make "-j$(nproc)"
+make install
+
+$CXX $CXXFLAGS -std=c++11 -I"$WORK/include/ImageMagick-7" "$SRC/encoder_list.cc" \
+    -o "$WORK/encoder_list" \
+    -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 \
+    "$WORK/lib/libMagick++-7.Q16HDRI.a" \
+    "$WORK/lib/libMagickWand-7.Q16HDRI.a" \
+    "$WORK/lib/libMagickCore-7.Q16HDRI.a"
+
+for f in $SRC/*_fuzzer.cc; do
+    fuzzer=$(basename "$f" _fuzzer.cc)
+    # encoder_fuzzer is special
+    if [ "$fuzzer" = "encoder" ]; then
+        continue
+    fi
+    $CXX $CXXFLAGS -std=c++11 -I"$WORK/include/ImageMagick-7" \
+        "$f" -o "$OUT/${fuzzer}_fuzzer" \
+        -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 \
+        -lFuzzingEngine "$WORK/lib/libMagick++-7.Q16HDRI.a" \
+        "$WORK/lib/libMagickWand-7.Q16HDRI.a" "$WORK/lib/libMagickCore-7.Q16HDRI.a"
+    cp afl_testcases.zip "$OUT/${fuzzer}_seed_corpus.zip"
```

Comment:
Are we using the same seed corpus for each fuzz target? Does it makes sense, i.e. how many files from `afl_testcases.zip` are really **interesting** for a particular codec?

Reply:
We are using the same seed corpus, yes. Most of them probably aren't interesting as the afl corpus is JPEG, GIF, PNG, BMP, ICO, WebP, and TIFF. If you'd like me to remove the corpus from most of the targets for now I can.

Reply:
How large is the current seed corpus? I mean number of files and total size in bytes.

Reply:
18.6k files, 83MB

Reply:
Thanks @alex ! It's too much, given that most of the files are not relevant :( We have two options:

1) Proceed without seed corpus. Not the best option, but can be good enough for start. We can add proper seed corpus in follow-up commits

2) Group seed corpus into relevant directories. For example, if there is a fuzzer that accepts PNG images as an input, we need to extract all PNG images from `afl_testcases.zip` and put them into `%fuzz_target_accepting_PNG%_seed_corpus.zip`:
 2.A) it can be done manually once, and those "groups" have to be landed in the repo or somewhere else (e.g. libreoffice: https://github.com/google/oss-fuzz/blob/master/projects/libreoffice/Dockerfile#L25)
 2.B) there can be a script that finds relevant corpus files for each fuzz target and creates a separate `%fuzz_target%_seed_corpus.zip` for that (there is an example in ffmpeg project, but it needs more work: https://github.com/google/oss-fuzz/blob/master/projects/ffmpeg/build.sh#L211)